packages/datadog_grpc_interceptor/test/datadog_grpc_interceptor_test.dart

```diff
@@ -288,6 +296,48 @@ void main() {
       expect(spanInt, BigInt.from(0x3eb5c1bcb46ab916));
     });
 
+    test('truncates b3m headers and sets attributes', () async {
+      when(() => mockDatadog.isFirstPartyHost(any())).thenReturn(true);
+
+      final interceptor = DatadogGrpcInterceptor(mockDatadog, channel);
+
+      final stub = GreeterClient(channel, interceptors: [interceptor]);
+
+      // 3c9bdc4fffcb6efe8f19eead81ad506b truncated will be 0f19eead81ad506b or
```

Comment:
IMO we shouldn't be truncating values in SDKs, they should be fully processed.

Reply:
Ah, I must have misread the spec. I thought we were expected to truncate.